plugin/storage/memory/memory_test.go

```diff
@@ -523,3 +523,92 @@ func makeTestingSpan(traceID model.TraceID, suffix string) *model.Span {
 		StartTime: time.Unix(300, 0).UTC(),
 	}
 }
+
+// TestInferredServiceDependency tests if the service dependencies include an inferred service
+// when there is a client span without a corresponding server span.
+func TestInferredServiceDependency(t *testing.T) {
+	// Create a memory store
+	store := NewStore()
+
+	// Create a client span without a corresponding server span
+	clientSpan := &model.Span{
```

Comment:
this is rather simplistic fixture. I'd say you also want to test:
  * span.kind=client which is NOT a leaf (has a child)
  * leaf span that is NOT span.kind=client

Reply:
Modified the test to check above two scenarios as well.